cpp/src/parquet/file_reader.cc

```diff
@@ -365,6 +367,20 @@ class SerializedFile : public ParquetFileReader::Contents {
             ComputeColumnChunkRange(file_metadata_.get(), source_size_, row, col));
       }
     }
+#ifndef NDEBUG
+    if (!ranges.empty()) {
```

Comment:
Maybe it's a better place to put this check in. I'm on the high railway home, so I don't have enough time to check this carefully. Any idea is welcome